pytorch_lightning/trainer/trainer.py

```diff
@@ -791,41 +847,63 @@ def __test_using_best_weights(self, ckpt_path, test_dataloaders):
             model.load_state_dict(ckpt['state_dict'])
 
         # attach dataloaders
-        if test_dataloaders is not None:
-            self.data_connector.attach_dataloaders(model, test_dataloaders=test_dataloaders)
+        if dataloaders is not None:
```

Comment:
Let me know if you have a better idea of how to handle this 😅

Reply:
It can be simpler considering you only have two cases:

```python
kwargs = {'test_dataloaders' if stage == 'test' else 'val_dataloaders': dataloaders}
self.data_connector.attach_dataloaders(model, **kwargs)
```

Reply:
Nice, I could do that indeed. Thanks!